Userland/Libraries/LibJS/Runtime/TypedArray.cpp

```diff
@@ -93,6 +93,32 @@ static void initialize_typed_array_from_array_buffer(GlobalObject& global_object
     typed_array.set_byte_offset(offset);
     typed_array.set_array_length(new_byte_length.value() / element_size);
 }
+template<typename T, TypedArrayType type>
+static void initialize_typed_array_from_typed_array(GlobalObject& global_object, TypedArray<T, type>& typed_array_to, TypedArrayBase& typed_array_from)
+{
+    auto array_length = typed_array_from.array_length();
+    auto element_size = typed_array_to.element_size();
+
+    typed_array_to.set_array_length(array_length);
+    typed_array_to.set_viewed_array_buffer(ArrayBuffer::create(global_object, array_length * element_size));
```

Comment:
Looks like `array_length * element_size` can overflow here, no?

Reply:
yes, working on it now